#! /bin/sh -e

# DP: Apply fix to allow libgomp.so to build as a shared library

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

@DPATCH@
diff -urNad libgomp/configure.tgt libgomp/configure.tgt
--- libgomp/configure.tgt	2007-07-04 10:19:19.000000000 -0400
+++ libgomp/configure.tgt	2008-07-09 16:09:54.000000000 -0400
@@ -17,8 +17,8 @@
   case "${target}" in
 
     *-*-linux*)
-	XCFLAGS="${XCFLAGS} -ftls-model=initial-exec"
-	XLDFLAGS="${XLDFLAGS} -Wl,-z,nodlopen"
+#	XCFLAGS="${XCFLAGS} -ftls-model=initial-exec"
+#	XLDFLAGS="${XLDFLAGS} -Wl,-z,nodlopen"
 	;;
   esac
 fi
